example-get-started-experiments/generate.sh

```diff
@@ -67,56 +71,45 @@ tick
 git commit -m "Initialize DVC project"
 
 
-cp -r $HERE/code/data .
-git add data/.gitignore data/pool_data.dvc
+cp -r $HERE/code/datasets .
+git add datasets/.gitignore datasets/pool_data.dvc
 tick
 git commit -m "Add data"
 dvc pull
 
 
-cp -r $HERE/code/notebooks .
+cp -r $HERE/code/TrainSegModel.ipynb .
 git add .
 git commit -m "Add notebook using DVCLive"
 
+sudo apt-get update && sudo apt-get install ffmpeg libsm6 libxext6 -y
 pip install -r requirements.txt --extra-index-url https://download.pytorch.org/whl/cu118
 pip install jupyter
-jupyter nbconvert --execute 'notebooks/TrainSegModel.ipynb' --inplace
-# Apply best experiment
-BEST_EXP_ROW=$(dvc exp show --drop '.*' --keep 'Experiment|evaluate/dice_multi|base_lr' --csv --sort-by evaluate/dice_multi | tail -n 1)
-BEST_EXP_NAME=$(echo $BEST_EXP_ROW | cut -d, -f 1)
-BEST_EXP_BASE_LR=$(echo $BEST_EXP_ROW | cut -d, -f 3)
-dvc exp apply $BEST_EXP_NAME
+yolo settings datasets_dir="/workspaces/example-repos-dev/example-get-started-experiments/build/example-get-started-experiments/datasets/"
+yolo settings runs_dir="/workspaces/example-repos-dev/example-get-started-experiments/build/example-get-started-experiments/runs/"
+yolo settings weights_dir="/workspaces/example-repos-dev/example-get-started-experiments/build/example-get-started-experiments/weights/"
+jupyter nbconvert --execute 'TrainSegModel.ipynb' --inplace
 git add .
 tick
 git commit -m "Run notebook and apply best experiment"
 git tag -a "1-notebook-dvclive" -m "Experiment using Notebook"
-gto register results/train:pool-segmentation --version v1.0.0
-gto assign results/train:pool-segmentation --version v1.0.0 --stage dev
+gto register dvclive:best --version v0.1.0
+gto assign dvclive:best --version v0.1.0 --stage dev
 
 
 cp -r $HERE/code/src .
 cp $HERE/code/params.yaml .
-sed -e "s/base_lr: 0.01/base_lr: $BEST_EXP_BASE_LR/" -i".bkp" params.yaml
-rm params.yaml.bkp
 
-dvc stage add -n data_split \
-  -p base,data_split \
-  -d src/data_split.py -d data/pool_data \
-  -o data/train_data -o  data/test_data \
-  python src/data_split.py
+dvc stage add -n create_yolo_dataset \
+  -d src/create_yolo_dataset.py -d datasets/pool_data \
+  -o datasets/yolo_dataset/train -o  datasets/yolo_dataset/val \
+  "python src/create_yolo_dataset.py \${create_yolo_dataset}"
 
-dvc remove models/model.pkl.dvc
 dvc stage add -n train \
-  -p base,train \
-  -d src/train.py -d data/train_data \
-  -o models/model.pkl \
-  python src/train.py
-
-dvc stage add -n evaluate \
-  -p base,evaluate \
-  -d src/evaluate.py -d models/model.pkl -d data/test_data \
-  python src/evaluate.py
+  -d src/train.py -d datasets/yolo_dataset/train -d datasets/yolo_dataset/val \
+  "python src/train.py \${train}"
```

Comment:
Feels odd to not have the model as an output of the stage here to me.

Also, why do have train and val datasets as separate deps/outs instead of tracking the whole dataset as one?

Reply:
> Feels odd to not have the model as an output of the stage here to me.

I can make the changes for that. I know odd but couldn't find a reason why it matters (maybe I should add additional stage)

> Also, why do have train and val datasets as separate deps/outs instead of tracking the whole dataset as one?

So in Studio you can use Data columns to show the number of files separately.

Reply:
> I can make the changes for that. I know odd but couldn't find a reason why it matters (maybe I should add additional stage)

Does https://github.com/iterative/dvclive/pull/631 break anything here since it skips caching if `_inside_dvc_exp=True`?

Reply:
> Does https://github.com/iterative/dvclive/pull/631 break anything here since it skips caching if _inside_dvc_exp=True?

Because the .dvc file is created at notebook stage, I believe there is no difference because DVC will (dvc) commit the changes at the end